themes/docs-new/layouts/shortcodes/unified_mode.md

```diff
@@ -0,0 +1,192 @@
+
+In Chef Infra Client 18.x, `unified_mode true` will become the default behavior.
+
+|Chef Infra Client  | Unified Mode                 |
+|-------------------|-----------------------------|
+| 18.x (2022)       | Default: `unified_mode true` |
+| 17.x              | Default: `unified_mode false`|
+| 16.x              | Default: `unified_mode false`|
+| 15.3 and higher   | Default: `unified_mode false`|
+| 15.0–15.2        | Not available               |
+| 14.14 and higher  | Default: `unified_mode false`|
+| Lower than 14.14  | Not available               |
+
+## Enable Unified Mode
+
+Enable unified mode by adding the `unified_mode true` in the resource declaration. You can upgrade most custom resources to use unified mode without additional work other than testing and validation.
+
+```ruby
+# enable unified mode
+unified_mode true
+
+provides :myresource
+
+actions :run do
+  [...]
+end
+
+```
+
+## Troubleshoot Unified Mode
+
+Unified mode changes the execution of a custom resource to run in one phase, in the order that the code is written, from first line of the code to the last. Custom resources designed to use two phases may need modification. These fall into three general types:
+
+- Resources with changes to internal sub-resources
+- Resources with actions on later resources
+- Resources that rely on the out-of-order execution
+
+When designing a custom resource for unified mode:
+
+- Declare a resource first and then declare actions on it
+- Write resources in run-time order
+
+### Resources with Changes to Internal Sub-resources
+
+Some resources are designed to create and edit other sub-resources as part of the resource declaration. In unified mode, the Chef Infra Client parses a resource codeblock that creates or edits a sub-resource and immediately and tries to apply that change, even though the sub-resource does not yet exist. This results in the execution of an incomplete resource.
+
+For example, with unified mode enabled, this code from the _dhcp cookbook_ is designed to create and edit a shared `dhcp_subnet` resource does not work as expected:
+
+```ruby
+# 'edit_resource' results in an incomple subresource
+sr = edit_resource(:dhcp_subnet, "#{new_resource.name}_sharedsubnet_#{subnet}") do
+  owner new_resource.owner
+  group new_resource.group
+  mode new_resource.mode
+
+  ip_version new_resource.ip_version
+  conf_dir new_resource.conf_dir
+  shared_network true
+end
+
+properties.each do |property, value|
+  sr.send(property, value)
+end
+```
+
+To fix resources that change sub-resources during their declaration, you can:
+
+- Apply properties in the codeblock (preferred)
+- Run the resource explicitly (not preferred)
+
+#### Apply Properties in the Codeblock
+
+This pattern declares the sub-resource in one codeblock and then changes it in the next codeblock. This is the preferred pattern in unified mode because all resources execute in order at compile time.
```

Comment:
```suggestion
This pattern declares the sub-resource in one code block and then changes it in the next code block. This is the preferred pattern in unified mode because all resources execute in order at compile time.
```